application/src/main/java/com/sanction/thunder/authentication/Key.java

```diff
@@ -1,30 +1,30 @@
 package com.sanction.thunder.authentication;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 
+import java.security.Principal;
+
 import static com.google.common.base.Preconditions.checkNotNull;
 
-public class Key {
+public class Key implements Principal {
 
   @JsonProperty("application")
-  private final String application;
+  private final String name;
 
   @JsonProperty("secret")
   private final String secret;
 
-  @JsonCreator
-  public Key(@JsonProperty("application") String application,
+  public Key(@JsonProperty("application") String name,
              @JsonProperty("secret") String secret) {
-    this.application = checkNotNull(application);
+    this.name = checkNotNull(name);
     this.secret = checkNotNull(secret);
   }
 
   @JsonProperty("application")
```

Comment:
Do you think you could try removing these `@JsonProperty` tags and see if it still works? If it does then let's keep them out. Otherwise make sure you change the `@JsonProperty("application")` to `@JsonProperty("name")`.